modules/test/portlet-container-test/test/integration/com/liferay/portlet/container/test/RequestTest.java

```diff
@@ -0,0 +1,798 @@
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portlet.container.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.TransactionalTestRule;
+import com.liferay.portal.kernel.util.HttpUtil;
+import com.liferay.portal.kernel.util.MapUtil;
+import com.liferay.portal.kernel.util.PropsKeys;
+import com.liferay.portal.kernel.util.ReflectionUtil;
+import com.liferay.portal.security.auth.AuthTokenWhitelistUtil;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.util.PropsValues;
+import com.liferay.portlet.PortletURLFactoryUtil;
+import com.liferay.portlet.PortletURLImpl;
+import com.liferay.portlet.SecurityPortletContainerWrapper;
+import com.liferay.util.Encryptor;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import java.lang.reflect.Field;
+
+import java.util.Collections;
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.portlet.PortletContext;
+import javax.portlet.PortletException;
+import javax.portlet.PortletRequest;
+import javax.portlet.PortletRequestDispatcher;
+import javax.portlet.PortletURL;
+import javax.portlet.RenderRequest;
+import javax.portlet.RenderResponse;
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+import javax.portlet.WindowState;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+
+/**
+ * @author Raymond Augé
+ */
+@RunWith(Arquillian.class)
+public class RequestTest extends BaseTestCase {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(), TransactionalTestRule.INSTANCE);
+
+	@Before
+	@Override
+	public void setUp() throws Exception {
+		super.setUp();
+
+		map = new ConcurrentHashMap<>();
+		requestTestPortlet = new RequestTestPortlet(map);
+		properties = new Hashtable<>();
+	}
+
+	@After
+	@Override
+	public void tearDown() throws Exception {
+		super.tearDown();
+
+		map = null;
+		requestTestPortlet = null;
+		properties = null;
+	}
+
+	@Test
+	public void testActionRequest_AUTH_TOKEN_CHECK_ENABLED() throws Exception {
+		Field field = ReflectionUtil.getDeclaredField(
+			PropsValues.class, "AUTH_TOKEN_CHECK_ENABLED");
+
+		field = ReflectionUtil.unfinalField(field);
```

Comment:
ReflectionUtil.getDeclaredField() does unfinalField(field) already, this is not needed.